test/e2e/e2e.go

```diff
@@ -558,16 +560,18 @@ func getApiAddress() string {
 }
 
 // IsGatewayModeLocal returns true if the gateway mode is local
-func IsGatewayModeLocal() bool {
-	anno, err := e2ekubectl.RunKubectl("default", "get", "node", "ovn-control-plane", "-o", "template", "--template={{.metadata.annotations}}")
-	if err != nil {
-		framework.Logf("Error getting annotations: %v", err)
-		return false
-	}
-	framework.Logf("Annotations received: %s", anno)
-	isLocal := strings.Contains(anno, "local")
-	framework.Logf("IsGatewayModeLocal returning: %v", isLocal)
-	return isLocal
+func IsGatewayModeLocal(cs kubernetes.Interface) bool {
+	node, err := e2enode.GetRandomReadySchedulableNode(context.TODO(), cs)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	l3ConfigRaw, ok := node.Annotations[util.OvnNodeL3GatewayConfig]
+	if !ok {
+		framework.Failf("Failed to find L3 gateway config for Node %q", node.Name)
+	}
+	var l3Config *util.L3GatewayConfig
+	if err = l3Config.UnmarshalJSON([]byte(l3ConfigRaw)); err != nil {
+		framework.Failf("Failed unmarshall Node %q L3 gateway config: %v", node.Name, err)
+	}
+	return l3Config.Mode == config.GatewayModeLocal
```

Comment:
```suggestion
	var l3Config *util.L3GatewayConfig
	if err = l3Config.UnmarshalJSON([]byte(l3ConfigRaw)); err != nil {
		framework.Failf("Failed unmarshall Node %q L3 gateway config: %v", node.Name, err)
	}
	gomega.Expect(l3Config.UnmarshalJSON([]byte(l3ConfigRaw))).To(Success())
	return l3Config.Mode == config.GatewayModeLocal
```

Reply:
Ahh my sugestion is wrong I mean 
```suggestion
	var l3Config *util.L3GatewayConfig
	Expect(l3Config.UnmarshalJSON([]byte(l3ConfigRaw))).To(Success("should unmarshall Node %q L3 gateway config", node.Name)
	return l3Config.Mode == config.GatewayModeLocal
```